### ISSUE TITLE
TypeErrors after executing "run dev" script block the build

### ISSUE THREAD

**Issue by lgortm** [vuefront/vuefront#1]
I installed vuefront with npx (on Windows 7, node v10.15.0). Backend is: woocommerce on localhost + VueFront CMS Connect App plugin. The Command "npm run dev" results in the following error: 

```
TypeError
Cannot read property 'code' of undefined
node_modules/vuefront/lib/store/common/language.js:125: 
```
![error](https://user-images.githubusercontent.com/29202779/59745075-3f74ed80-9274-11e9-81c6-e1078e7f61f4.png)

npm run generate also results in errors:
[generate-errors.txt](https://github.com/vuefront/vuefront/files/3304886/generate-errors.txt)

**Comment by lgortm** [comment]
I wasn't able to run dev also on Linux Mint (Virtual box) , there were also type errors. Vuefront url in CLI prompt was set to remote server (Woo commerce) and cors to true.
![linux-mint](https://user-images.githubusercontent.com/29202779/59753207-ec0a9b80-9283-11e9-944e-2ce88520e355.png)

**Comment by dmitriyzhuk** [comment]
Hi, please provide the Wordpress and woocommerce versions

**Comment by lgortm** [comment]
Hi,

**Localhost params:**  

Wordpress 5.2.2
WooCommerce version: 3.6.4
PHP Version 7.2.11
HTTP_HOST aptekaback.test (virtualhost using https://laragon.org/)

**Remote server params:**  

WordPress 5.0.4
WooCommerce version: 3.5.4
PHP Version 7.1.21

I also attach the CMS Connect URL
https://maszyny.develoop.pl/wp-content/plugins/d_vuefront/index.php

**Comment by dmitriyzhuk** [comment]
Hi, the issue was in create-vuefront-app.  I have just updated the NPM package with version 0.1.5 

please give it another try by running the VueFront installation from the start 

```bash
yarn create vuefront-app <project-name>
```

Please confirm if this helped.

**Comment by lgortm** [comment]
Thanks,
I comfirm that it works. Nuxt starts without errors, but only on Linux .

On Windows, i couldn't even generate project.
It seems to install dependencies after i anwser prompt questions, but always ends in error and the project folder isn't generated.
I tried with default CMD, git-bash without success. Yesterday, i was able to generate Vuefront project on Windows using npx.


Windows 7.

node v10.16.0

**YARN v1.16.0**


```
warning ..\..\package.json: No license field
error Command "lint" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

```

**NPX**
npm v 6.9.0

```
npm ERR! missing script: lint

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user\AppData\Roaming\npm-cache\_logs\2019-06-20T17_15_41_430Z-debug.log
```

**Comment by lgortm** [comment]
I made it work also on Windows. I copied the generated project from Linux, deleted node_modules, re-installed dependencies with yarn and everything works. As i've written above there is still a problem in generating new projects on Windows.

**Comment by nzxt** [comment]
I got the same error on
- Windows 10 1903
- Yarn 1.16.0
- PHP 7.3 x64
- Opencart 3.1.0.0_b or 3.0.3.2
... installing latest 3.0.3.1.v1.0.0.ocmod

Direct link shows me error: **message: "Syntax Error GraphQL (1:1) Unexpected <EOF> 1: ^ "**
and **Cannot read property 'code' of undefined** in language.js.
VueFront in modules in status Disabled (is it normal?)

Thanks! I'll try one more time..

**Comment by dmitriyzhuk** [comment]
The direct link will show this because GraphQL GET request should return an error. we will make a status page later on so it would give a better understanding. 

Please try with php version less then 7.3 (like 7.2) there is a limitation with the GraphQL library. we will add support for 7.3 and higher in the next version 

also please check your CMS Connect URL - is there s & or a ? symbol before the cors=true 

if you are on Wordpress, it should be ? (this bug we just fixed in the last update of vuefront/create-vuefront-app)

**Comment by nzxt** [comment]
Query is in good condition, there is '?' as I'm on OpenCart. Error disappeared when I turn PHP version from 7.3x64 to 7.2x64. Continue to dig in..
Is it possible to build VueFront in SPA/PWA without SSR, by simply switching to mode:'spa' in Nuxt, would it work?
Thanks! Great work.

**Comment by dmitriyzhuk** [comment]
Hi, yes, it also supports without SSR but there are limitations because e-commerce is very dynamic (stocks, different prices etc). 
Give it a go and let us know how we can improve it

**Comment by dmitriyzhuk** [comment]
I will close this issue. Feel free to create a new one.

**Comment by Alligatore3** [comment]
Hi @dmitriyzhuk 
This morning I tested this fantastic tool, but unfortunately I run into this issue.

I'm testing in my local's Ubuntu machine.

Here's some details:
> yarn --version => 1.16.0 
node --version => v.11.13.0
Wordpress => 5.2.2
WooCommerce version => 3.6.5
Server => Apache/2.4.29 (Ubuntu)
PHP --version => 7.2.19-0ubuntu0.18.04.1
PHP Post Max Size => 8 MB
MySQL --version => 5.7.26-0ubuntu0.18.04.1
Max upload dim. => 2 MB
CMS connect Link => http://localhost/wordpress/wp-content/plugins/d_vuefront/index.php

I run `yarn create vuefront-app <project-name>` twice because I thought it was a CORS problem (1^ time replying YES, 2^ NO).
Nothing changed.

I also attached my `yarn.lock` maybe it's usefull

Thanks, M.

[yarn.lock.zip](https://github.com/vuefront/vuefront/files/3431220/yarn.lock.zip)

**Comment by dmitriyzhuk** [comment]
Hi @Alligatore3 
Please try this latest release v1.1.0

https://github.com/vuefront/wordpress/releases

The CMS Connect app was a bit outdated. let me know if that helped you

**Comment by Alligatore3** [comment]
You rock @dmitriyzhuk 🤘🏻
It works!

**Comment by dertuerke** [comment]
I get this error with Wordpress and vuefront 1.1
```
{"errors":[{"message":"Syntax Error GraphQL (1:1) Unexpected \n\n1: \n ^\n","category":"graphql","locations":[{"line":1,"column":1}]}]}
```

**Comment by dmitriyzhuk** [comment]
Hi, yes, that is a normal response from a GraphQL API. basically, it says - you just made a GET request to us, but haven't provided any parameters so  I am returning an error to you. 

Just use this CMS Connect URL when launching your VueFront Web App via YARN (or NPM)

**Comment by dertuerke** [comment]
@dmitriyzhuk ok thanks.
But if i run `npm run dev` then i get this error in the terminal
`[GraphQL error]: Message: Cannot query field "home" on type "RootQueryType"., Location: [object Object], Path: undefined`

**Comment by Alligatore3** [comment]
@dertuerke Me too.
Same error with me also with `yarn run dev`

**Comment by Alligatore3** [comment]
@dmitriyzhuk 

> Just use this CMS Connect URL when launching your VueFront Web App via YARN (or NPM)

How?

**Comment by dmitriyzhuk** [comment]
@dertuerke Are you following the installation guide

```bash
yarn create vuefront-app <project-name>
```

if so, please let me know the following info:
php version
CMS 
nodejs v
Operation system

And I will check it localy

**Comment by dmitriyzhuk** [comment]
@Alligatore3 when you first run 

```bash
yarn create vuefront-app <project-name>
```

you will be asked to provide the CMS connect URL in the command line. just copy paste it there and finish the installation. 

after that you can run

```bash
yarn dev
```

**Comment by Alligatore3** [comment]
@dmitriyzhuk 
That's exactly what I did.

I tried 2 ways:

1. from localhost wordpress installation with MAMP.

2. remote from my hosting with wordpress.


**In both ways** I got this from terminal's console:
`WARN  [GraphQL error]: Message: Cannot query field "home" on type "RootQueryType"., Location: [object Object], Path: undefined`

And this from browser's console:
![local](https://user-images.githubusercontent.com/8971094/62423092-20fc7180-b6bd-11e9-9a3c-a87df0fcb142.jpg)

I've also switched nuxt mode from `universal` to `spa` but nothing changed.
Got graphQL error: 
`message: "Cannot query field "home" on type "RootQueryType"."`

**Comment by mirkosaugo** [comment]
I've the same error @Alligatore3 . Hey @dmitriyzhuk this is my configuration:
- Node  v10.15.3
- MacOs Mojave
- yarn v1.10.1
- WP : v5.2.2
- Woocommerce : v3.6.5
- Vuefront: v1.1 (but in the admin always print 1.0, is it normal?)

**Comment by dmitriyzhuk** [comment]
Guys, @mirkosaugo @Alligatore3 

so actually you are doing everything right, it is just that we have updated the WebApp and are still pending updates to Wordpress release. There are some breaking points in the versions.

The issue with wordpress is that we are working to submit it to wordpress plugin repo, which puts some heavy limitations on us. 

What you can do, is use the MASTER branch from vuefront/wordpress to test it. it should work. we will soon update the release as well. 

Sorry for the inconvenience.

**Comment by Alligatore3** [comment]
@mirkosaugo KALMATY
> Vuefront: v1.1 (but in the admin always print 1.0, is it normal?)

Yes it is. [Have a look here](https://github.com/vuefront/wordpress/pull/1)


@dmitriyzhuk thanks for support.
> What you can do, is use the MASTER branch from vuefront/wordpress to test it.

I tried, in the same both ways as I mentioned before.
No success, but different errors.
**I think it's hard to report due to different environments, maybe a docker image should me more stable, BTW here's the reports**
1. from http://localhost/wordpress I got
`Cannot read property 'code' of undefined` from browser and ` WARN  [Network error]: ServerParseError: Unexpected token < in JSON at position 0` from terminal.
2. from http://mattiazanella.it/develop/wp/ I got blank page, no errors from terminal but in browser's console:
 `error [Vue warn]: The client-side rendered virtual DOM tree is not matching server-rendered content. This is likely caused by incorrect HTML markup, for example nesting block-level elements inside <p>, or missing <tbody>. Bailing hydration and performing full client-side render.`

If this is becoming too verbose thread let me know, we can move another side.
Thanks again.


M. 🐊
